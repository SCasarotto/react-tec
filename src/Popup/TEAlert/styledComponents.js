import styled from 'styled-components'

import TEPopup from './../TEPopup'
import TEButton from './../../Form/TEButton'

export const Popup = styled(TEPopup)`
	${(props) => {
		return `
            & > div{
                width: 90%;
                max-width: 600px;
                margin: 100px auto;
            }
        `
	}}
`
export const Title = styled.h4`
	${(props) => {
		const { theme } = props
		return `
            position: relative;
            width: 100%;
            text-align: center;
            font-size: 24px;
            margin-bottom: 20px;
            color: ${theme.darkerGray};
            padding: 30px 30px 0px 30px;
        `
	}}
`
export const Message = styled.p`
	${(props) => {
		const { theme } = props
		return `
            position: relative;
            width: 100%;
            text-align: center;
            font-size: 18px;
            line-height: 1.25;
            margin-bottom: 30px;
            padding: 0px 40px;
            color: ${theme.darkerGray};
        `
	}}
`
export const Button = styled(TEButton)`
	${(props) => {
		const { theme } = props
		return `
        border-top: 1px solid ${theme.primary};
        border-right: none;
        border-bottom: none;
        border-left: none;
        line-height: 2;
        border-top-left-radius: 0px;
        border-top-right-radius: 0px;
        border-bottom-left-radius: 20px;
        border-bottom-right-radius: 20px;
        transition: border 0.2s ease-in, color 0.2s ease-in, background-color 0.2s ease-in;

        :active {
            color: ${theme.white};
            background-color: ${theme.primary};
            border-top: 1px solid ${theme.primary};
            border-right: none;
            border-bottom: none;
            border-left: none;
        }
        :hover {
            color: ${theme.white};
            background-color: ${theme.primary};
            border-top: 1px solid ${theme.primary};
            border-right: none;
            border-bottom: none;
            border-left: none;
        }
        `
	}}
`
