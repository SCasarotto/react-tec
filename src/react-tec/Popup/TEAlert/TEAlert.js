//
//TE Version 0.2.0
//

import React, { Component } from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'

import styles from './styles'

import TEPopup from './../TEPopup'
import TEButton from './../../Form/TEButton'

class TEAlert extends Component {
	render() {
		const { title, message, onClick, buttonTitle, children, visible } = this.props

		return (
			<TEPopup visible={visible} contentStyles={styles.content}>
				<div style={styles.title}>{title}</div>
				<div style={styles.message}>{message}</div>
				{children}
				<TEButton onClick={onClick} style={styles.button}>
					{buttonTitle}
				</TEButton>
			</TEPopup>
		)
	}
}

TEAlert.propTypes = {
	title: PropTypes.string,
	message: PropTypes.string,
	onClick: PropTypes.func.isRequired,
	buttonTitle: PropTypes.string,
}

TEAlert.defaultProps = {
	buttonTitle: 'Okay',
}

export default Radium(TEAlert)
