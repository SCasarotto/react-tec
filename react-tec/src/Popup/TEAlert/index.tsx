import React from 'react'

import { Popup, Title, Message, Button } from './styledComponents'
import { TEPopupProps } from 'Popup/TEPopup'

export interface TEAlertProps extends TEPopupProps {
	title?: string
	message?: string
	onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void
	buttonTitle?: string
}
const TEAlert: React.FC<TEAlertProps> = (props) => {
	const {
		title,
		message,
		onClick,
		buttonTitle = 'Okay',
		visible,
		className = '',
		children,
	} = props

	return (
		<Popup visible={visible} className={`TEAlert ${className}`}>
			<Title className="TEAlertTitle">{title}</Title>
			<Message className="TEAlertMessage">{message}</Message>
			{children}
			<Button onClick={onClick} className="TEAlertButton">
				{buttonTitle}
			</Button>
		</Popup>
	)
}

export default TEAlert
