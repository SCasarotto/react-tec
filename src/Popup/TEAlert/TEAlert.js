//
//TE Version 0.3.0
//

import React from 'react'
import PropTypes from 'prop-types'

import { Popup, Title, Message, Button } from './styledComponents'

const TEAlert = (props) => {
	const { title, message, onClick, buttonTitle, visible, className, children } = props

	return (
		<Popup visible={visible} className={className}>
			<Title className='TEAlertTitle'>{title}</Title>
			<Message className='TEAlertMessage'>{message}</Message>
			{children}
			<Button onClick={onClick} className='TEAlertButton'>
				{buttonTitle}
			</Button>
		</Popup>
	)
}

TEAlert.propTypes = {
	title: PropTypes.string,
	message: PropTypes.string,
	onClick: PropTypes.func.isRequired,
	buttonTitle: PropTypes.string,
}

TEAlert.defaultProps = {
	buttonTitle: 'Okay',
}

export default TEAlert
