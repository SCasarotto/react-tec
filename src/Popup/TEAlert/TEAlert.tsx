import React from 'react'

import { Popup, Title, Message, Button } from './styledComponents'
import { TEPopupProps } from './../TEPopup'

export interface TEAlertProps extends TEPopupProps {
	title?: string
	message?: string
	onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void
	buttonTitle?: string
}
/**
 *
 * TEAlert is a simple alert popup typically used to display error or success messages.
 *
 */
export const TEAlert: React.FC<TEAlertProps> = (props) => {
	const {
		title,
		message,
		onClick,
		buttonTitle = 'Okay',
		visible,
		className = '',
		children,
	} = props

	return (
		<Popup visible={visible} className={`TEAlert ${className}`}>
			{title && <Title className='TEAlertTitle'>{title}</Title>}
			{message && <Message className='TEAlertMessage'>{message}</Message>}
			{children}
			<Button onClick={onClick} className='TEAlertButton'>
				{buttonTitle}
			</Button>
		</Popup>
	)
}
