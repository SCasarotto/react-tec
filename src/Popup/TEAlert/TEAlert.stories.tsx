import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TEAlert, TEAlertProps } from './TEAlert'

export default {
	title: 'Popup/TEAlert',
	component: TEAlert,
	argTypes: {},
} as Meta

const Template: Story<TEAlertProps> = (args) => <TEAlert {...args} />

export const Default = Template.bind({})
Default.args = {
	title: 'TEAlert!',
	message: 'You have been alerted to stay alert.',
	visible: true,
	children: <p style={{ textAlign: 'center' }}>Some extra components passed into the body</p>,
}
