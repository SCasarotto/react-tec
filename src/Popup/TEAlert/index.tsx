import React from 'react'

import { Popup, Title, Message, Button } from './styledComponents'
import { TEPopupProps } from 'Popup/TEPopup'

export interface TEAlertProps extends TEPopupProps {
	title?: string
	message?: string
	onClick?(event: React.MouseEvent<HTMLButtonElement, MouseEvent>): void
	buttonTitle?: string
}
export const TEAlert: React.FC<TEAlertProps> = (props) => {
	const {
		title,
		message,
		onClick,
		buttonTitle = 'Okay',
		visible,
		className = '',
		children,
	} = props

	return (
		<Popup visible={visible} className={`TEAlert ${className}`}>
			{title && <Title className="TEAlertTitle">{title}</Title>}
			{message && <Message className="TEAlertMessage">{message}</Message>}
			{children}
			<Button onClick={onClick} className="TEAlertButton">
				{buttonTitle}
			</Button>
		</Popup>
	)
}
