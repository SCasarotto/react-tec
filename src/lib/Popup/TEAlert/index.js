import TEAlert from './TEAlert'

export default TEAlert
