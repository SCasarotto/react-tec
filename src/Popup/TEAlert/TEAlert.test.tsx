import React from 'react'
import { render } from '@testing-library/react'
import { TEAlert } from './TEAlert'

describe('TEAlert', () => {
	// Renders Without Error
	it('matches snapshot', () => {
		const { asFragment } = render(<TEAlert visible />)
		expect(asFragment()).toMatchSnapshot()
	})

	// Class Name
	it('accepts custom className', () => {
		const { container } = render(<TEAlert visible className='myClassName' />)
		expect(container.firstChild).toHaveClass('myClassName')
	})
})
