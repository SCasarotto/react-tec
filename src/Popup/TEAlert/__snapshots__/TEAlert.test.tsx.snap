// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TEAlert matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx kmiImc TEPopup sc-hKFyIo eFASRH TEAlert "
  >
    <div
      class="sc-gtssRu jqgoxm TEPopupContent"
    >
      <button
        class="sc-dlnjPT MgKjE TEButton sc-gKAblj ilNIXJ TEAlertButton"
        type="button"
      >
        Okay
      </button>
    </div>
  </div>
</DocumentFragment>
`;
